submission/persistence/src/main/kotlin/ac/uk/ebi/biostd/persistence/service/filesystem/FilesService.kt

```diff
@@ -12,47 +14,65 @@ import ebi.ac.uk.extended.model.allFiles
 import ebi.ac.uk.extended.model.allReferencedFiles
 import ebi.ac.uk.io.FileUtils
 import ebi.ac.uk.io.FileUtils.copyOrReplaceFile
-import ebi.ac.uk.io.FileUtils.deleteFolder
+import ebi.ac.uk.io.FileUtils.deleteFile
+import ebi.ac.uk.io.FileUtils.getOrCreateFolder
 import ebi.ac.uk.io.FileUtils.moveFile
-import ebi.ac.uk.io.FileUtils.reCreateDirectory
+import ebi.ac.uk.io.FileUtils.reCreateFolder
+import ebi.ac.uk.io.ONLY_USER
 import ebi.ac.uk.paths.FILES_PATH
 import ebi.ac.uk.paths.SubmissionFolderResolver
 import java.io.File
 import java.nio.file.Path
+import java.nio.file.attribute.PosixFilePermission
+import java.nio.file.attribute.PosixFilePermissions
+
+private val READ_ONLY_GROUP: Set<PosixFilePermission> = PosixFilePermissions.fromString("rwxr-x---")
+private val ALL_READ: Set<PosixFilePermission> = PosixFilePermissions.fromString("rwxr-xr-x")
 
 class FilesService(
     private val folderResolver: SubmissionFolderResolver,
     private val serializationService: SerializationService
 ) {
+
     fun persistSubmissionFiles(submission: ExtSubmission, mode: FileMode) {
+        val permissions = permissions(submission.released)
         val submissionPath = folderResolver.getSubmissionFolder(submission.relPath)
-        generateOutputFiles(submission, submissionPath)
+        generateFiles(submission, submissionPath, permissions)
+
         when (mode) {
-            FileMode.MOVE -> processFiles(submission, submissionPath, ::move)
-            FileMode.COPY -> processFiles(submission, submissionPath, ::copy)
+            MOVE -> processFiles(submission, submissionPath) { file, path -> move(file, path, permissions) }
+            COPY -> processFiles(submission, submissionPath) { extFile, file -> copy(extFile, file, permissions) }
         }
     }
 
-    private fun generateOutputFiles(submission: ExtSubmission, submissionPath: Path) {
+    private fun permissions(released: Boolean): Set<PosixFilePermission> {
+        return if (released) ALL_READ else return READ_ONLY_GROUP
+    }
+
+    private fun generateFiles(
+        submission: ExtSubmission,
+        submissionPath: Path,
+        permissions: Set<PosixFilePermission>
+    ) {
         val simpleSubmission = submission.toSimpleSubmission()
 
-        generateOutputFiles(simpleSubmission, submissionPath, submission.accNo)
-        submission.allFileList.forEach { generateOutputFiles(it.toFilesTable(), submissionPath, it.fileName) }
+        generateFiles(simpleSubmission, submissionPath, submission.accNo, permissions)
+        submission.allFileList.forEach { generateFiles(it.toFilesTable(), submissionPath, it.fileName, permissions) }
     }
 
-    // TODO add file list content validation to integration tests
-    // TODO add special character folders/files names test
-    // TODO Test temporally folder already existing
-    // TODO we need to remove also pagetab files as only FILES are clean right now
-    // TODO add integration test for file list within subsections
```

Comment:
Do we have this test already or is it no longer necessary?

Reply:
yes unit test in the file utils